File: classes01/python/homography_book.py
import cv2
import numpy as np

im_src=cv2.imread("../../data/images/book2.jpg")
pts_src=np.float32([[141, 131], [480, 159], [493, 630],[64, 601]])

im_dst=cv2.imread("../../data/images/book1.jpg")
pts_dst=np.float32([[318, 256],[534, 372],[316, 670],[73, 473]])

ho,status=cv2.findHomography(pts_src,pts_dst)

im_out=cv2.warpPerspective(im_src,ho,(im_src.shape[1],im_dst.shape[0]))

print(im_src.shape)

cv2.imshow("Source Image", im_src)
cv2.imshow("Destination", im_dst)
cv2.imshow("Warp", im_out)

cv2.waitKey(0)
cv2.destroyAllWindows()

File: classes01/cpp/homography_book.cpp
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main(void)
{
    // 原始图像
    Mat im_src=imread("../../data/images/book2.jpg");
    vector<Point2f> pts_src;
    pts_src.push_back(Point2f(141, 131));
    pts_src.push_back(Point2f(480, 159));
    pts_src.push_back(Point2f(493, 630));
    pts_src.push_back(Point2f(64, 601));

    // 目标图像
    Mat im_dst=imread("../../data/images/book1.jpg");
    vector<Point2f> pts_dst;
    pts_dst.push_back(Point2f(318, 256));
    pts_dst.push_back(Point2f(534, 372));
    pts_dst.push_back(Point2f(316, 670));
    pts_dst.push_back(Point2f(73, 473));

    // 计算Homography矩阵
    Mat ho=findHomography(pts_src,pts_dst);

    // 输入图像
    Mat im_out;
    warpPerspective(im_src,im_out,ho,im_src.size());

    // 显示图片
    imshow("Source", im_src);
    imshow("Destination", im_dst);
    imshow("Warped Source Image", im_out);

    waitKey(0);
    destroyAllWindows();


}